plugins/linux-pipewire/pipewire.c

```diff
@@ -1012,58 +986,97 @@ static void on_param_changed_cb(void *user_data, uint32_t id,
 	obs_pipewire *obs_pw = obs_pw_stream->obs_pw;
 	struct spa_pod_builder pod_builder;
 	const struct spa_pod *params[5];
-	const char *format_name;
+	struct spa_rectangle size;
+	struct spa_fraction rate;
 	uint32_t n_params = 0;
 	uint32_t buffer_types;
-	uint32_t output_flags;
 	uint8_t params_buffer[1024];
 	int result;
 
 	if (!param || id != SPA_PARAM_Format)
 		return;
 
+	spa_zero(obs_pw_stream->format);
+
 	result = spa_format_parse(param, &obs_pw_stream->format.media_type,
 				  &obs_pw_stream->format.media_subtype);
 	if (result < 0)
 		return;
 
-	if (obs_pw_stream->format.media_type != SPA_MEDIA_TYPE_video ||
-	    obs_pw_stream->format.media_subtype != SPA_MEDIA_SUBTYPE_raw)
+	if (obs_pw_stream->format.media_type != SPA_MEDIA_TYPE_video)
 		return;
 
-	spa_format_video_raw_parse(param, &obs_pw_stream->format.info.raw);
+	switch (obs_pw_stream->format.media_subtype) {
```

Comment:
Unfortunately, not using the enum type in the switch might lead to unwanted omissions.

```suggestion
	switch ((enum spa_media_subtype)obs_pw_stream->format.media_subtype) {
```

As an example `SPA_MEDIA_SUBTYPE_h264` is missing in the switch, and we can't know if it's normal.